catalog/dags/data_refresh/distributed_reindex.py

```diff
@@ -0,0 +1,425 @@
+"""
+# Distributed Reindex TaskGroup
+
+This module contains the Airflow tasks used for orchestrating the reindexing of records from the temporary tables in the downstream (API) database, into a new Elasticsearch index. Reindexing is performed on a fleet of indexer worker EC2 instances, with instance creation and termination managed by Airflow.
+"""
+
+import logging
+import math
+from textwrap import dedent
+from typing import Sequence
+from urllib.parse import urlparse
+
+from airflow import settings
+from airflow.decorators import task, task_group
+from airflow.exceptions import AirflowSkipException
+from airflow.models.connection import Connection
+from airflow.providers.amazon.aws.hooks.ec2 import EC2Hook
+from airflow.providers.http.operators.http import HttpOperator
+from airflow.providers.http.sensors.http import HttpSensor
+from airflow.sensors.base import PokeReturnValue
+from airflow.utils.trigger_rule import TriggerRule
+from requests import Response
+
+from common.constants import (
+    AWS_CONN_ID,
+    OPENLEDGER_API_CONN_ID,
+    PRODUCTION,
+    Environment,
+)
+from common.sql import PGExecuteQueryOperator, single_value
+from data_refresh.constants import INDEXER_LAUNCH_TEMPLATES, INDEXER_WORKER_COUNTS
+from data_refresh.data_refresh_types import DataRefreshConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+WORKER_CONN_ID = "indexer_worker_{worker_id}_http_{environment}"
+
+
+class TempConnectionHTTPOperator(HttpOperator):
```

Comment:
I actually _meant_ for temp to stand for temporary, as these are temporary connections that are created and destroyed when the EC2 instance we're connecting to is destroyed! 🤣 However it does happen to also be templated, so if we want to pull these out to be reusable then changing the name to `TemplatedConnectionHTTPOperator` is a better name because it's not so specific to this use case.

Reply:
Oh jeez 😆 Yea if you're okay with that, I think pulling them out would be ideal!